src/Features/Core/Portable/AliasAmbiguousType/AbstractAliasAmbiguousTypeCodeFixProvider.cs

```diff
@@ -43,8 +44,10 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             {
                 var addImportService = document.GetRequiredLanguageService<IAddImportsService>();
                 var syntaxGenerator = document.GetRequiredLanguageService<SyntaxGenerator>();
+                var codeGenerator = document.GetRequiredLanguageService<ICodeGenerationService>();
                 var compilation = semanticModel.Compilation;
-                var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var preferences = await CodeGenerationPreferences.FromDocumentAsync(document, cancellationToken).ConfigureAwait(false);
```

Comment:
interesting.  we have CodeGenerationPreferences.FromDocumentAsync  *and* CSharpCodeGenerationPreferences.FromDocumentAsync?

Reply:
when should the latter be called?

Reply:
The former is can be called from lang agnostic code, the latter from C#. It's the same modulo casts.